src/test/java/htsjdk/samtools/filter/NotPrimaryAlignmentFilterTest.java

```diff
@@ -0,0 +1,108 @@
+
+        FilteringSamIterator filteringSamIterator = new FilteringSamIterator(builder.getRecords().iterator(),
+                new NotPrimaryAlignmentFilter());
+
+        Assert.assertEquals(filteringSamIterator.hasNext(),false);
+    }
+
+    @Test
+    public void testPrimaryRecords() {
+        SAMRecordSetBuilder builder = new SAMRecordSetBuilder();
+        int i = 0;
+        for (boolean record1Unmapped : new boolean[]{true, false}) {
+            for (boolean record2Unmapped : new boolean[]{true, false}) {
+                for (boolean record1Strand : new boolean[]{true, false}) {
+                    for (boolean record2Strand : new boolean[]{true, false}) {
+
+                        builder.addPair("pair" + i, 0, 10, 30,
+                                record1Unmapped, record2Unmapped,
+                                null, null,
+                                record1Strand, record2Strand,
+                                false, false,
+                                10);
+                        builder.addFrag("frag" + i++, 0, 10,
+                                record1Unmapped,
+                                record2Strand,
+                                null,null,
+                                10,false);
+                    }
+                }
+            }
+        }
+
+        FilteringSamIterator filteringSamIterator = new FilteringSamIterator(builder.getRecords().iterator(),
+                new NotPrimaryAlignmentFilter());
+
+        Assert.assertEquals(filteringSamIterator.stream().count(),i*3);
+    }
+
+
+
+    @Test
+    public void testSupplementaryRecords() {
+        SAMRecordSetBuilder builder = new SAMRecordSetBuilder();
+        int i = 0;
+        for (boolean record1Unmapped : new boolean[]{true, false}) {
+            for (boolean record2Unmapped : new boolean[]{true, false}) {
+                for (boolean record1Strand : new boolean[]{true, false}) {
+                    for (boolean record2Strand : new boolean[]{true, false}) {
+
+                        builder.addPair("pair" + i, 0, 10, 30,
+                                record1Unmapped, record2Unmapped,
+                                null, null,
+                                record1Strand, record2Strand,
+                                false, false,
+                                10).stream().forEach(r->r.setSupplementaryAlignmentFlag(true));
+                        builder.addFrag("frag" + i++, 0, 10,
+                                record1Unmapped,
+                                record2Strand,
+                                null,null,
+                                10,false).setSupplementaryAlignmentFlag(true);
+                    }
+                }
+            }
+        }
+
+        FilteringSamIterator filteringSamIterator = new FilteringSamIterator(builder.getRecords().iterator(),
+                new NotPrimaryAlignmentFilter());
+
+        Assert.assertEquals(filteringSamIterator.stream().count(),i*3);
```

Comment:
Can you add a comment above each assert explaining why (for example) `i * 3` is the expected value?